lib/cisco_node_utils/vxlan_vtep_vni.rb

```diff
@@ -63,8 +71,34 @@ def set_args_keys(hash={})
     end
     # rubocop:enable Style/AccessorMethodNamefor
 
+    def create_with_associate_vrf?
+      !@set_args[:vrf].eql?('')
+    end
+
+    def destroy_existing(key)
+      getargs = { name: @name, vni: @vni, state: '' }
+      # rubocop:disable Style/GuardClause
+      if config_get('vxlan_vtep', key, getargs)
+        key.eql?('vni_with_vrf') ? vrf = 'associate-vrf' : vrf = ''
+        getargs[:vrf] = vrf
+        getargs[:state] = 'no'
+        config_set('vxlan_vtep', 'vni', getargs)
+      end
+      # rubocop:enable Style/GuardClause
+    end
+
     def create
+      # The configuration for this resource can be either of the following:
+      # - member nve 5000
+      # - member nve 5000 associate-vrf
+      # They are mutually exclusive and one must be removed before the other
+      # can be configured.
       set_args_keys(state: '')
+      if create_with_associate_vrf?
+        destroy_existing('vni_without_vrf')
+      else
+        destroy_existing('vni_with_vrf')
+      end
       config_set('vxlan_vtep', 'vni', @set_args)
```

Comment:
I actually can't.... :) I need to call `set_args_keys` before I call `create_with_associate_vrf?` so that @set_args has stuff in it.